### ISSUE TITLE
Lines between paired data points for stripplot (and maybe others)

### ISSUE THREAD

**Issue by guptadivyansh** [mwaskom/seaborn#3363]
The situation arises quite often that the same sample is measured twice, and thus data is 'paired'. Each pair is usually plotted as a line connecting the two conditions. However, I could not find any support for such a plot in sns. My attempt was to use a stripplot (and boxplot) for the dots and then manually add the lines with plt.

Just sns:

```
sns.boxplot(data=temporalFits, x='state', y='alpha')
# make grouped stripplot
sns.stripplot(data=temporalFits, x='state', y='alpha',
                alpha=0.5, jitter=False,
                color='grey', legend= False)
```
![a48cd0a3-bb4e-4991-ad86-f6d8cd276ce2](https://github.com/mwaskom/seaborn/assets/9139390/568e5f16-5b9f-4e2b-8f24-30f7b5e88c81)

I added the following lines to get the plot I needed.

```
#plot a line connecting the points for each neuron
for neuron in temporalFits['neuron'].unique():
    plt.plot([0,1], temporalFits[temporalFits['neuron']==neuron]['alpha'], color = 'black', alpha = 0.2)
```

![0f6a6ed1-a001-4b21-a059-c394c97dc712](https://github.com/mwaskom/seaborn/assets/9139390/a8fdf4a3-7fcb-4aa6-8b9b-0ce3956fa433)

The only issue being that it cannot work properly with jitter = True.

Such data arise often enough (for example in medical research), that I think sns should support it. 

Does anyone know if sns has any system for dealing with paired data? If there is sufficient interest, I could try digging into the internals to make it work.

**Comment by mwaskom** [comment]
This is something that can be accomplished with `lineplot`

```python
fmri = sns.load_dataset("fmri").query("event == 'stim' and timepoint == 8")
sns.boxplot(fmri, x="region", y="signal")
sns.lineplot(fmri, x="region", y="signal", units="subject", estimator=None, marker="o", color=".3", mew=0)
```
![image](https://github.com/mwaskom/seaborn/assets/315810/abae6d7c-8bc3-4ca9-a241-719df1060e0c)

I'd recommend making this kind of plot like this, which is less cluttered (note that `gap=0.2` will be introduced in seaborn v0.13+)

```python
fmri = sns.load_dataset("fmri").query("event == 'stim' and timepoint == 8")
sns.boxplot(fmri, x="region", y="signal", hue="region", dodge=True, gap=.2)
sns.lineplot(fmri, x="region", y="signal", units="subject", estimator=None, marker="o", color=".3", mew=0)
```
![image](https://github.com/mwaskom/seaborn/assets/315810/ebd8033d-a5a2-4424-9f60-3d665ce93667)

**Comment by hmassalha** [comment]
Thanks for your detailed explanation. I would like to level up the previous question... 
I have groped box plots (see attached) in a catplot format. I am interested in linking the medians of the boxes within each group (manually added cyan line in the below image). I tried using pointplot but always ended up linking medians of the same group across all levels of x-axis... any suggestions would be much appreciated. 

![image](https://github.com/mwaskom/seaborn/assets/10478803/7a114754-1aff-44ee-963f-70011ec22009)
